.github/workflows/generic.yml

```diff
@@ -52,34 +56,47 @@ jobs:
       - name: git checkout
         uses: actions/checkout@v4
 
+      - name: Run sccache
+        uses: mozilla-actions/sccache-action@v0.0.4
+
       - name: Install Protoc
         uses: arduino/setup-protoc@9b1ee5b22b0a3f1feb8c2ff99b32c89b3c3191e9 # v2.0.0
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: install rust toolchain
         uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          components: clippy
```

Comment:
Are we sure this is needed? Clippy was and is working afaik without this.

Reply:
Yes, I just found out this from official documentation:

> If you're using rustup to install and manage your Rust toolchains, Clippy is usually already installed. In that case you can skip this chapter and go to the [Usage](https://doc.rust-lang.org/clippy/usage.html) chapter.

Reply:
Yeah, I don't think it is actually needed. I just thought it makes it more explicit(which I like) and it seems to be required if we decide to use a GH action for clippy like [this one](https://github.com/giraffate/clippy-action), which would show clippy warnings more visually.
But I don't have a strong opinion here, so happy to remove it if you think it is better.

Reply:
the problem is, then we should also add `cargo-fmt` and other components (if we are using anything else) to be consistent. Seeing `clippy` but not seeing `fmt` confuses me a bit :D 

I rather have them not listed here, but if you prefer to see them explicitly, I suggest we also add `fmt` to the components. 

And for [clippy action](https://github.com/giraffate/clippy-action), I don't think it is required, clippy is installed regardless whether it's stated explicitly or not.

This is all nitpicking, not a blocker for the PR, the choice is yours :)

Reply:
I agree, and decided to make `rustmft` explicit as well, so that it is clear what components we are using and we are sure they are always available(regardless of the architecture we use and whether it is stable or nightly). Thanks for the feedback!